core/src/main/java/org/chodavarapu/datamill/http/Client.java

```diff
@@ -61,108 +75,155 @@ protected URLConnection createConnection(String uri) throws IOException {
         if (uriParameters != null && uriParameters.size() > 0) {
             uri = uriBuilder.build(uri, uriParameters);
         }
+        URIBuilder uriBuilder = new URIBuilder();
+
+
+        int firstSlash = uri.indexOf('/');
+        uriBuilder.setScheme(uri.substring(0, firstSlash - 1));
+
+        String uriNoScheme = uri.substring(firstSlash + 1, uri.length());
+        uriBuilder.setHost(uriNoScheme.substring(1));
 
-        uri = appendQueryParameters(uri, queryParameters);
+        URI anURI = null;
+        try {
+            anURI = appendQueryParameters(uriBuilder, queryParameters);
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+        }
 
+        final URI theURI = anURI;
         final String composedUri = uri;
 
         return Async.fromCallable(() -> {
-            URLConnection urlConnection = createConnection(composedUri);
-            HttpURLConnection httpConnection = (HttpURLConnection) urlConnection;
+            PipedOutputStream pipedOutputStream = null;
+            PipedInputStream pipedInputStream = null;
+
+            try {
+
+
+                CloseableHttpClient httpclient = HttpClients.createDefault();
```

Comment:
Getting rid of finally blocks for them makes code look cleaner, but now we're initialising the streams even for request types not sending an entity, and thus not needing the streams. Not sure this is right tbh..